Alerts component for Appeals v2.

markgreenburg opened the pull request:
Components that together render a list of alerts - one for each object in the `appeal.attributes.alerts` array. Also introduces a helper function into `helpers-v2` that maps content to each alert type.

To-Do:
[x] - Wire up Alerts and Alert to each other, Status page
[x] - Finish logic for both components (prop passing, conditional styling based on alert type)
[x] - Make sure everything looks right at different viewport sizes
[x] - Everything accompanied by unit tests

**Full**
<img width="750" alt="screen shot 2017-11-27 at 12 11 57 pm" src="https://user-images.githubusercontent.com/24251447/33282379-168f143e-d36e-11e7-98a5-3b6e015900cc.png">


**Mobile**
<img width="403" alt="screen shot 2017-11-27 at 12 11 43 pm" src="https://user-images.githubusercontent.com/24251447/33282387-1d2ce38e-d36e-11e7-8264-e0bad8b53189.png">

markgreenburg committed:
Begin work on Alerts for Appeals v2.

markgreenburg committed:
Refactor html into Alerts component, wire up data

markgreenburg committed:
Refactor individual alert into own component

markgreenburg committed:
Remove .only unit test restrictions

markgreenburg committed:
Update Alert test with new props

markgreenburg committed:
Add cssClass property to default case in getAlertContent helper

gnakm commented:
I wasn't fully aware of the new look for alert messages (My bads, I should have checked). I was hoping that the icons in the alert messages would align with the grey arrow. 

Example:
![image](https://user-images.githubusercontent.com/13420618/33279760-b911d648-d36d-11e7-9288-57a54282c2e4.png)

However, this is not going to work because it's going to break the grid on the left. 

Can we move the error message so that the highlighted bar is centered aligned with the arrow? 
Example:
![image](https://user-images.githubusercontent.com/13420618/33280203-502120c4-d36f-11e7-9b7f-6c85007d981b.png)

![image](https://user-images.githubusercontent.com/13420618/33280225-5f17cc22-d36f-11e7-97e6-d904cc698c23.png)

markgreenburg committed:
Align alerts with Timeline

markgreenburg commented:
Excellent point @gnakm - I just updated the alignment to have the vertical bars match up with the Timeline - I updated the screenshots in the first comment.

markgreenburg commented:
Sorry, looks like I didn't hit 'save' after uploading new screenshots - should be good now

cvalarida reviewed (commented):
My only real concern is the alert object structure inference--I don't think it'll look quite like that. Should be a pretty quick fix.

The rest of it looks good, though. 👍

gnakm reviewed (approved):
lgtm! Thanks!

markgreenburg commented on src/js/claims-status/containers/AppealsV2StatusPage.jsx:
```diff
@@ -49,9 +49,10 @@ AppealsV2StatusPage.defaultProps = {
       status: {
         type: '',
-        details: {}
+        details: {},
```
No, just habit. Trying to break that habit for consistency with the rest of the team but sometimes I forget :)

markgreenburg commented on src/js/claims-status/utils/helpers.js:
```diff
@@ -283,8 +283,18 @@ export const mockData = {
+            type: 'warning',
+            date: '09-21-2017',
+            details: {
+              type: 'waiting_on_action'
```
Yeah, I bet you're right. Chris G mentioned that shape will be consistent with other parts of the API, so I'm assuming it'll look like Status does. I've simplified this and re-worked a bit, though we may need to tweak again once the API is live.

markgreenburg commented on src/js/claims-status/utils/appeals-v2-helpers.jsx:
```diff
@@ -246,3 +258,61 @@ export function getNextEvents(currentStatus) {
+          representative, the ${details.representative}, to complete an
+          informal hearing presentation (IHP). Please contact your
+          representative for more information.`,
+        cssClass: 'usa-alert-warning',
```
I'm just going to hardcode a `cssClass` property value for each `alert.type` for now and get rid of the STYLE_TYPES enum for now. We'll see if this needs to change once we can use the API.

markgreenburg commented:
Going to address a few cleanup items in a separate PR
